public/js/ajax-airline-handler.js

```diff
@@ -0,0 +1,151 @@
+$('form').submit(function(event){
+    event.preventDefault();
+})
+
+const createAirline = () => {
+    fetch('api/airlines', {
+        method: 'POST',
+        headers: {
+            'Content-type': 'application/x-www-form-urlencoded',
+            'X-Requested-With': 'XMLHttpRequest',
+            'X-CSRF-TOKEN': $('meta[name="csrf-token"]').attr('content')
+        },
+        body: new URLSearchParams($('#add-airline-form').serialize())
+    })
+
+    .then(response => {
+        if (response.ok) {
+            $('#name-error').text('');
+            $('#description-error').text('');
+            document.forms["add-airline-form"].reset();
+
+            return response.json();
+
+        } else {
+            return response.json().then(data => {
+                handleValidationErrors(data.errors);
+            });
+        }
+    })
+    .then(result => {
+        loadTable();
```

Comment:
Does this get executed after you do this?
```
return response.json().then(data => {
    handleValidationErrors(data.errors);
});
```
I believe it does, but I could be wrong. This function should not be executed if an error occurs.